compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -296,8 +303,24 @@ BLangStatementExpression desugar(BLangQueryExpr queryExpr, SymbolEnv env,
             BType refType = Types.getReferredType(queryExpr.getBType());
             if (TypeTags.isXMLTypeTag(refType.tag) || (refType.tag == TypeTags.UNION
                     && ((BUnionType) refType).getMemberTypes().stream().allMatch(memType ->
-                                    TypeTags.isXMLTypeTag(Types.getReferredType(memType).tag)))) {
-                result = getStreamFunctionVariableRef(queryBlock, QUERY_TO_XML_FUNCTION, Lists.of(streamRef), pos);
+                    TypeTags.isXMLTypeTag(Types.getReferredType(memType).tag == TypeTags.INTERSECTION ?
+                            ((BIntersectionType) Types.getReferredType(memType)).effectiveType.tag :
+                            Types.getReferredType(memType).tag)))) {
+                BLangLiteral isReadonly = ASTBuilderUtil.createLiteral(null, symTable.booleanType, false);
```

Comment:
We can probably do 
```
            BLangLiteral isReadonly = ASTBuilderUtil.createLiteral(pos, symTable.booleanType,
                    Symbols.isFlagOn(queryExpr.getBType().flags, Flags.READONLY));
```
here also can get rid of the first if clause.

Reply:
Will update. Thanks.